js/src/controls/Control.ts

```diff
@@ -20,34 +24,35 @@ export class LeafletControlModel extends widgets.WidgetModel {
   }
 }
 
-export class LeafletControlView extends utils.LeafletWidgetView {
-  initialize(parameters) {
+export abstract class LeafletControlView extends LeafletWidgetView {
+  map_view: any;
+  obj: L.Control;
+
+  initialize(
+    parameters: WidgetView.IInitializeParameters<LeafletControlModel>
+  ): void {
     super.initialize(parameters);
     this.map_view = this.options.map_view;
   }
 
-  render() {
-    return Promise.resolve(this.create_obj()).then(() => {
-      this.leaflet_events();
-      this.model_events();
-    });
+  async render() {
+    await Promise.resolve(this.create_obj());
```

Comment:
I wonder, can we do this? I guess not with an abstract method?

```suggestion
    await this.create_obj();
```